Enterprise-CMCS/eAPD
# Tbolt/4411 Fixes issue where PUTs against /states/:stateId/affiliations/:id could assign invalid roles

Resolves #4411 

### Description
Adds a check to limit State Admins to only update affiliations to either `eAPD State Contractor` or `eAPD State Staff` and that Fed Admins can only update affiliations to `eAPD State Admin`

### Significant changes or possible side effects
Updating certifications uses the same logic as affiliations and was updated to support this new check

### Automated test cases written

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| a state admin | updating an affiliation to eAPD System Admin (or similar elevated role) | fails updating role | tap |
| a fed admin | updating an affiliation to eAPD State Staff (or non-State Admin role) | fails updating role | tap |
| a fed admin | updating an affiliation match without a corresponding certification  | fails updating role | tap |

### Steps to manually verify this change
1. Use an HTTP client (Postman, Insomnia, Paw, etc...)
2. Login to APD as a state admin (that has been approved as state admin)
3. Grab the JWT token from your cookies
4. Add that as a Bearer token in your HTTP client
5. Hit the update affiliations endpoint (ex. `PUT http://localhost:8081/states/ak/affiliations/1`)
6. Attempt to update to an invalid role ID
7. Verify a `400` error is returned with an error message
8. Repeat above steps as a fed admin attempting to assign a role that's not State Admin

### This pull request is ready to code review when

      tests, refactor
      screenreader, text scaling) OR an exemption is documented

### This pull request is ready to test when


### This pull request is ready to review when the QA has


### This pull request can be merged when
